pcsx2/Gif.cpp

```diff
@@ -31,8 +31,109 @@ static u32 gscycles = 0, prevcycles = 0, mfifocycles = 0;
 static u32 gifqwc = 0;
 static bool gifmfifoirq = false;
 
+__aligned16 GIF_Fifo gif_fifo;
+
+static __fi void GifDMAInt(int cycles) {
+	if (dmacRegs.ctrl.MFD == MFD_GIF) {
+		if (!(cpuRegs.interrupt & (1 << DMAC_MFIFO_GIF)) || cpuRegs.eCycle[DMAC_MFIFO_GIF] < (u32)cycles)
+		{
+			CPU_INT(DMAC_MFIFO_GIF, cycles);
+		}
+	} else if (!(cpuRegs.interrupt & (1 << DMAC_GIF)) || cpuRegs.eCycle[DMAC_GIF] < (u32)cycles)
+	{
+		CPU_INT(DMAC_GIF, cycles);
+	}
+}
 static __fi void clearFIFOstuff(bool full) {
-	CSRreg.FIFO = full ? CSR_FIFO_FULL : CSR_FIFO_EMPTY;
+		CSRreg.FIFO = full ? CSR_FIFO_FULL : CSR_FIFO_EMPTY;
```

Comment:
It was actually arcum's i believe, he was doing a cleanup and did that.

I'd have to look at the code properly to decide that, I was trying to avoid changing the old code in this PR as I didn't want to confuse things further.